### DISCUSSION TITLE
URL structure (routes)

### DISCUSSION THREAD

**Discussion by silviot** [librocco/librocco#118] [Ideas]
The root
--------

The root should serve as a redirection point.
When a user requests `/` the code will determine a route to go to, and redirect the user accordingly. Initially it will always redirect to `/dashboard`. When we'll implement authentication it will redirect you to `/login` if you're not authenticated, and to the dashboard otherwise.

The dashboard
-------------

As the application grows its feature should be visible on a dashboard.
For now only the inventory management will be shown.
The minimal implementation, for now, is a single button that links to `/inventory`.

The inventory management
------------------------

For now this is the only section of the application. In the future there will be more.
Thus all views in this section will live under `/inventory`.

The `/inventory/stock` view should show all books present in any warehouse.
`/inventory` should redirect to `/inventory/stock`.

Pagination will not change the URL. Search will change the URL: `/inventory/stock?q=medieval+history` will be the URL after you search "medieval history".

Different warehouses can be accessed with URLs such as `/inventory/stock/whjsXbPJNqc8wqGgX4youWE9`.

When creating a new note a random id will be assigned to it (let's say `note7Jud1GnNpnbNGTu13Q`).
The note will be accessible at `/inventory/stock/notes/note7Jud1GnNpnbNGTu13Q`.

**Comment by ikusteu** [comment]
I like this routing, I would only add a subroute to notes, to differentiate between inbound and outbound notes. This will be convenient as the page rendered can get all the necessary data from the route (note view, note id and note type).

Additionally, I would separate the stock and note paths, resulting in something like this:
- `/inventory/stock` - shows the entire inventory
- `/inventory/stock/[warehouseName]` - shows the inventory for a particular warehouse

- `/inventory/inbound/` - shows all inbound notes (on side nav) and nothing in the table
- `/inventory/inbound/[noteId]` - shows the same page as above, only with the table populated with the data for a particular note

- `/inventory/outbound` - see `inventory/inbound` ☝️ 
- `/inventory/outbound/[noteId]` - see `inventory/inbound/[noteId]` ☝️ 

Following this route structure, we can implement the pages with the following folder structure:
- `inventory/stock/[[warehouse]]/+page.svelte`
- `inventory/inbound/[[note]]/+page.svelte`
- `inventory/outbound/[[note]]/+page.svelte`
Double bracket segment ☝️ represents an optional route param so `inventory/stock/[[warehouse]]/+page.svelte` would catch both `/inventory/stock` as well as `/inventory/stock/[warehouse]`, which works quite nicely as it's essentially the same page with presence of `warehouse` param making a difference as to what will be shown (if anything) in the table section. The same is applicable to `inbound`/`outbound` routes

**Reply by ikusteu** [reply]
Idk about 404, but I think it can simply fall back to `/inventory`

**Reply by silviot** [reply]
+1 for everything, except the 404: I think it's better to clearly state the requested resource is not found than redirecting to a working page. We can include a link to `/inventory` in the 404 page to make user's lives easier.
